spork/math.janet

```diff
@@ -1475,22 +1494,26 @@
       (var res (qr R1))
       (set Q1 (res :Q))
       (set R1 (res :R))
-      (var res^ (qr (trans-m R1)))
+      (var res^ (qr (trans R1)))
       (set Q2 (res^ :Q))
       (set R2 (res^ :R))
-      (set R1 (trans-m R2))
+      (set R1 (trans R2))
       (set U (matmul U Q1))
       (set V (matmul V Q2)))) 
   {:U U
    :S R1
    :V V})
 
 (defn m-approx=
-  "Evaluates a matrix (list of row vectors) for equivalence within epsilon."
-  [m1 m2]
-  # TODO: 
-  (let [v1 (apply array/concat m1)
-        v2 (apply array/concat m2)
+  "Compares two matrices of equal size for equivalence within epsilon."
+  [m1 m2 &opt tolerance]
+  (let [v1 (squeeze m1)
+        v2 (squeeze m2)
         b  (map approx-eq v1 v2)]
     (and (= (length v1) (length v2))
-         (every? b)))) 
+         (every? b))))
+
+(let [m3 @[@[1 2 3] @[4 5 6] @[7 8 9]]]
+  (assert (m-approx= (matmul m3 (ident (rows m3)))
+                     m3)
+          "matmul identity left: this test succeeds here but fails in suite-math.janet"))
```

Comment:
Any idea why?

Reply:
Fixed. Caused by stateful computation due to join-rows not being idempotent (`array/concat` mutates first argument).